crates/apollo-parser/test_data/lexer/err/0014_international.graphql

```diff
@@ -0,0 +1,7 @@
+  mutation {
+      messageSender(
+          message: "some ok outer string "Tráfico" more ok outer string",
+      ) {
+        delivered
+  }
+}
```

Comment:
the issue with this test is that a string value here must not have quotation marks. Only block string values can have (escaped) quotation marks. So `Tráfico` gets registered as a named token, and _named tokens_ can [only be literal letters and numbers](https://spec.graphql.org/October2021/#Name). What was the utf-8 issue that you run into here?

I'll remove this test as it doesn't relate to the PR; we can add a fix once I understand the issue you're facing.

Reply:
This test was added to cover a regression I found from real queries. The problem was that the lexer was capturing utf-8 characters for `Name` tokens and this triggers a panic in the parser.

So with the byte sequence `"some ok outer string "Tráfico" more ok outer string"` I ended up with something like:
```rust
String "some ok outer string "
Ident Tráfico # triggers panic
String " more ok outer string"
```

I fixed the issue by using [`prev_str` instead of `current_str`](https://github.com/allancalix/apollo-rs/blob/38e09b082a5a3d8706e841f8dfce3372a4b9f3e8/crates/apollo-parser/src/lexer/mod.rs#L269) in the branch. This matches the previous lexer's behavior by creating something like:
```rust
String "some ok outer string "
Ident Tr
Ident fico
String " more ok outer string"

Error á
```

@lrlna The alternate way of handling this that might make more sense is to produce a token of `Ident Tráfico`. The line below sometimes triggers a panic when trying to index into string slices because depending on the character `name[1..]` may fall _inside the boundaries of a utf-8 character_ and not in between separate characters.
https://github.com/apollographql/apollo-rs/blob/efa365e40847913f2a7aa870206c529a796f8c64/crates/apollo-parser/src/parser/grammar/name.rs#L22